### PR TITLE
[FIVE-166]-UpsertDTO

### PR DESCRIPTION
**Pull request by gperelstein**
-Se eliminaron referencias innecesarias en el ArtifactDTO y en el ArtifactTypeDTO
-Se crearon los DTO's ArtifactUpsertDTO y CategoryUpsertDTO
-Se modifico el BaseApiController para recibir dos clases genéricas TPost y TPut
-Se agregaron al AutoMapper los mappeos de ArtifactUpsertDTO con Models.Artifact y CategoryUpsertDTO con Models.Category

### CODE REVIEW

File: FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs [hunk 1]
```diff
@@ -24,10 +24,15 @@ public AutoMapperProfile()
                 .ReverseMap();
             CreateMap<Category, Categories.CategoryDTO>()
                 .ReverseMap();
+            CreateMap<Categories.CategoryDTO, Category>();
             CreateMap<Artifact, ArtifactDTO>()
                 .ReverseMap()
                 .ForMember(dest => dest.ArtifactTypeId, opt => opt.MapFrom(src => src.ArtifactType.Id))
                 .ForMember(dest => dest.ProjectId, opt => opt.MapFrom(src => src.ProjectId));
+            CreateMap<Artifact, ArtifactUpsertDTO>()
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs:32]: You should never map from Artifact to insert/update DTOs. Only from input to business `ArtifactUpsertDTO -> Artifact`.

File: FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs [hunk 2]
```diff
@@ -24,10 +24,15 @@ public AutoMapperProfile()
...
+                .ReverseMap()
+                .ForMember(dest => dest.ArtifactTypeId, opt => opt.MapFrom(src => src.ArtifactTypeId))
+                .ForMember(dest => dest.ProjectId, opt => opt.MapFrom(src => src.ProjectId));
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs:35]: When props have the same name there's no need to specify how to resolve them.

File: FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs (hunk 1 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs]: Fixed it!

File: FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs (hunk 2 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web.Dtos/AutoMapperProfile.cs]: Fixed it!

**Review by matiasbeckerle — APPROVED** [review]

**Review by matiasbeckerle — APPROVED** [review]

**Review by gonzalorom — APPROVED** [review]
Looks all good for me.

**Pull request merged**